streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -31,60 +31,71 @@
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.regex.Pattern;
 
+import static org.apache.kafka.streams.kstream.internals.StreamsTopologyGraphImpl.TOPOLOGY_ROOT;
+import static org.apache.kafka.streams.kstream.internals.TopologyNodeType.GLOBAL_KTABLE;
+import static org.apache.kafka.streams.kstream.internals.TopologyNodeType.KTABLE;
+import static org.apache.kafka.streams.kstream.internals.TopologyNodeType.SOURCE;
+
+@SuppressWarnings("unchecked")
 public class InternalStreamsBuilder implements InternalNameProvider {
 
     final InternalTopologyBuilder internalTopologyBuilder;
 
+    private final StreamsTopologyGraph topologyGraph;
+    private boolean topologyNeedsBuilding = true;
+
     private final AtomicInteger index = new AtomicInteger(0);
 
     public InternalStreamsBuilder(final InternalTopologyBuilder internalTopologyBuilder) {
         this.internalTopologyBuilder = internalTopologyBuilder;
+        this.topologyGraph = new StreamsTopologyGraphImpl();
     }
 
     public <K, V> KStream<K, V> stream(final Collection<String> topics,
                                        final ConsumedInternal<K, V> consumed) {
         final String name = newProcessorName(KStreamImpl.SOURCE_NAME);
 
-        internalTopologyBuilder.addSource(consumed.offsetResetPolicy(),
-                                          name,
-                                          consumed.timestampExtractor(),
-                                          consumed.keyDeserializer(),
-                                          consumed.valueDeserializer(),
-                                          topics.toArray(new String[topics.size()]));
+        final ProcessDetails processDetails = ProcessDetails.builder().withConsumed(consumed).withSourceTopics(topics).build();
+        final StreamsGraphNode node = new StreamsGraphNode(name, SOURCE, false, processDetails, TOPOLOGY_ROOT);
+
+        topologyGraph.addNode(node);
```

Comment:
Maybe use the `addNode()` available on this class for consistency? (applies a few times in this file)